localstack/services/apigateway/apigateway_listener.py

```diff
@@ -370,7 +373,9 @@ def apply_request_parameters(uri: str, integration: Dict[str, Any], path_params:
             request_param_value = f"method.request.path.{key}"
             if request_parameters.get(request_param_key, None) == request_param_value:
                 uri = uri.replace(f"{{{key}}}", path_params[key])
-    return uri
+
+    # include the request query to the uri
+    return f"{uri}?{urlencode(query_params)}"
```

Comment:
Wondering if we can rely on `uri` not containing a `?`, or whether the integration URI itself may contain query parameters as well. 

Perhaps the logic would be a bit more robust (also against future changes) if we introduce a small utility function like `add_query_params_to_url(..)` in `utils/http_utils.py` that adds query parameters to an existing arbitrary URL (which may or may not contain query params). (We could also add an option whether to overwrite or expand any existing query params in this function.)

Reply:
Thanks, agreed!